src/main/java/org/jabref/logic/exporter/GlobalSaveManager.java

```diff
@@ -0,0 +1,89 @@
+import org.jabref.model.FieldChange;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntryTypesManager;
+import org.jabref.preferences.PreferencesService;
+
+public class GlobalSaveManager {
+
+    private static Set<GlobalSaveManager> runningInstances = new HashSet<>();
+
+    private final DelayTaskThrottler<Set<Character>> throttler = new DelayTaskThrottler<>(1500);
+    private final BibDatabaseContext bibDatabaseContext;
+
+    private final StateManager stateManager;
+
+    private final PreferencesService preferencesService;
+
+    private GlobalSaveManager(StateManager stateManager, PreferencesService preferencesService, BibEntryTypesManager entryTypesManager) {
+        this.bibDatabaseContext=  stateManager.getActiveDatabase().get();
+        this.stateManager = stateManager;
+        this.preferencesService = preferencesService;
+
+    }
+
+    public static void shutdown(BibDatabaseContext context) {
+        runningInstances.stream().filter(instance -> instance.bibDatabaseContext == context).forEach(GlobalSaveManager::shutdown);
+        runningInstances.removeIf(instance -> instance.bibDatabaseContext == context);
+    }
+
+    public static GlobalSaveManager create(StateManager stateManager, PreferencesService preferencesService, BibEntryTypesManager entryTypesManager) {
+        GlobalSaveManager saveAction = new GlobalSaveManager(stateManager, preferencesService, entryTypesManager);
+        runningInstances.add(saveAction);
+        return saveAction;
+    }
+
+    private void shutdown() {
+        this.throttler.shutdown();
+
+    }
+
+    public Future<Set<Character>> save(Path file, boolean selectedOnly, Charset encoding, SavePreferences.DatabaseSaveType saveType, Consumer<List<FieldChange>> consumeFieldChanges) {
+        return throttler.scheduleTask(() -> saveThrottled(file, selectedOnly, encoding, saveType, consumeFieldChanges));
+    }
+
+    private Set<Character> saveThrottled(Path file, boolean selectedOnly, Charset encoding, SavePreferences.DatabaseSaveType saveType, Consumer<List<FieldChange>> consumeFieldChanges) throws SaveException {
+
+        SavePreferences savePrefs = this.preferencesService.getSavePreferences()
+                                                    .withEncoding(encoding)
+                                                    .withSaveType(saveType);
+
+        try (AtomicFileWriter fileWriter = new AtomicFileWriter(file, savePrefs.getEncoding(), savePrefs.shouldMakeBackup())) {
+            BibtexDatabaseWriter databaseWriter = new BibtexDatabaseWriter(fileWriter, savePrefs, Globals.entryTypesManager);
+
+            if (selectedOnly) {
+                databaseWriter.savePartOfDatabase(this.bibDatabaseContext, this.stateManager.getSelectedEntries());
+            } else {
+                databaseWriter.saveDatabase(this.bibDatabaseContext);
+            }
+
+            consumeFieldChanges.accept(databaseWriter.getSaveActionsFieldChanges());
```

Comment:
I would introduce a class (record? ;-)) as the return value of this method, with fields `encodingProblems` and `fieldChanges`. Should make the code a bit cleaner